search/django/weedid/views.py

```diff
@@ -103,18 +105,61 @@ def upload_info(request):
 
 
 def upload_list(request):
-    upload_list = map(
-        lambda queryEntity: {
-            "name": queryEntity.metadata["info"][0]["name"]
-            if "name" in queryEntity.metadata["info"][0]
-            else "",
-            "upload_id": queryEntity.upload_id,
-            "upload_date": str(queryEntity.date),
-            "contributor": queryEntity.user.username,
-        },
-        Dataset.objects.filter(status="C"),
-    )
-    return HttpResponse(json.dumps(list(upload_list)))
+    user = request.user
+    if user and user.is_authenticated:
+        upload_list = map(
+            retrieve_listing_info,
+            Dataset.objects.filter(status="C"),
+        )
+        return HttpResponse(json.dumps(list(upload_list)))
+    else:
+        return HttpResponseForbidden("You dont have access to proceed")
+
+
+def awaiting_list(request):
+    user = request.user
+    if user and user.is_authenticated and user.is_staff:
+        awaiting_list = map(
+            retrieve_listing_info,
+            Dataset.objects.filter(status="AR"),
+        )
+        return HttpResponse(json.dumps(list(awaiting_list)))
+    else:
+        return HttpResponseForbidden("You dont have access to proceed")
+
+
+def dataset_approve(request, dataset_id):
+    user = request.user
+    if user and dataset_id and user.is_authenticated and user.is_staff:
+        upload_entity = Dataset.objects.get(upload_id=dataset_id, status="AR")
+        if upload_entity:
+            weedcoco_path = os.path.join(
+                REPOSITORY_DIR, str(dataset_id), "weedcoco.json"
+            )
+            update_index_and_thumbnails(weedcoco_path, dataset_id)
+            return HttpResponse("It has been approved")
+        else:
+            return HttpResponseNotAllowed("Dataset to be reviewed doesnt exist")
+    else:
+        return HttpResponseForbidden("You dont have access to proceed")
```

Comment:
Flat is better than nested?

```suggestion
    if not (user and dataset_id and user.is_authenticated and user.is_staff):
        return HttpResponseForbidden("You dont have access to proceed")
    upload_entity = Dataset.objects.get(upload_id=dataset_id, status="AR")
    if not upload_entity:
        return HttpResponseNotAllowed("Dataset to be reviewed doesnt exist")

    weedcoco_path = os.path.join(
        REPOSITORY_DIR, str(dataset_id), "weedcoco.json"
    )
    update_index_and_thumbnails(weedcoco_path, dataset_id)
    return HttpResponse("It has been approved")
```